packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -680,11 +680,38 @@ typedef InspectorSelectionChangedCallback = void Function();
 
 /// Structure to help reference count Dart objects referenced by a GUI tool
 /// using [WidgetInspectorService].
-class _InspectorReferenceData {
-  _InspectorReferenceData(this.object);
+@visibleForTesting
+class InspectorReferenceData {
+  /// Creates an instance of [InspectorReferenceData].
+  InspectorReferenceData(Object object) {
+    if (value is Record) {
+      throw StateError('Cannot wrap Record, because it is not supported by $WeakReference.');
```

Comment:
nit: why not just let the WeakReference support itself throw? There are other cases like some ffi classes that would probably although throw.